vdr/ambassador.go

```diff
@@ -99,12 +100,21 @@ func (n *ambassador) callback(tx dag.Transaction, payload []byte) error {
 		return fmt.Errorf("callback could not process new DID Document, DID Document integrity check failed: %w", err)
 	}
 
-	isUpdate, err := n.isUpdate(nextDIDDocument)
-	if err != nil {
-		return fmt.Errorf("callback could not process new DID Document, failed to resolve current DID Document: %w", err)
+	// we check the VDR if the document is already processed by using the transaction hash
+	sourceTX := tx.Ref()
+	doc, _, err := n.didStore.Resolve(nextDIDDocument.ID, &types.ResolveMetadata{
+		SourceTransaction: &sourceTX,
```

Comment:
is this also correct when documents are conflicted?

Reply:
If any of the sourcetransactions of the conflicted docs matches it is returned. It then becomes a conflicted document with 3 source transactions (if there were previous 2)